### PR TITLE
Adds snmp endpoints

### LINKED ISSUE #613: snmp endpoints
Endpoints for the snmp functionality should be added including /snmp, /snmp/users, /snmp/communities, and /snmp/traps

### PR DESCRIPTION
**Pull request by caphrim007**
Issues:
Fixes #613

Problem:
The api is missing functionality for manipulating the snmp endpoints

Analysis:
This change adds functionality for the snmp endpoints
- snmp
- snmp/communities
- snmp/traps
- snmp/users

Tests:
functional and unit

### CODE REVIEW

**Comment by wojtek0806** [comment]
looks good, however I would also add modify test to SNMP endpoint, as it supposed to support PATCH method, unless it does not and docs are incorrect.

File: test/functional/tm/sys/test_snmp.py [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+            request, mgmt_root, 'Common', 'modtest1'
+        )
+        original_dict = copy.copy(comm1.__dict__)
+        commName = 'communityName'
+        comm1.modify(communityName='footest1')
+        for k, v in original_dict.items():
+            if k != commName:
+                original_dict[k] = comm1.__dict__[k]
+            elif k == commName:
+                comm1.__dict__[k] == 'footest1'
```
Comment (wojtek0806) [inline test/functional/tm/sys/test_snmp.py]: should this not be something like:

```
assert comm1.__dict__[k] == 'footest1' 

```

**Comment by zancas** [comment]
@caphrim007 Is there anything blocking this from merging (other than review)?

File: test/functional/tm/sys/test_snmp.py (hunk 1 above)
Comment (zancas) [inline test/functional/tm/sys/test_snmp.py]: I think we should add `assert` here like @wojtek0806 suggests.

Comment (zancas) [inline test/functional/tm/sys/test_snmp.py]: @caphrim007 so I think all that remains to be done is to add "`assert`".

**Comment by zancas** [comment]
`assert LGTM`

**Pull request merged**